src/wazuh_db/wdb_global.c

```diff
@@ -971,79 +1065,100 @@ wdbc_result wdb_global_sync_agent_info_get(wdb_t *wdb, int* last_agent_id, char
-            snprintf(*output, WDB_MAX_RESPONSE_SIZE, "%s", "Cannot bind sql statement");
-            status = WDBC_ERROR;
-            break;
-        }
+    int stmts[] = {
+        WDB_STMT_GLOBAL_SYNC_REQ_FULL_GET,
+        WDB_STMT_GLOBAL_SYNC_REQ_STATUS_GET,
+        WDB_STMT_GLOBAL_SYNC_REQ_KEEPALIVE_GET
+    };
 
-        //Get agent info
-        cJSON* sql_agents_response = wdb_exec_stmt(agent_stmt);
-        if (sql_agents_response && sql_agents_response->child) {
-            cJSON* json_agent = sql_agents_response->child;
-            cJSON* json_id = cJSON_GetObjectItem(json_agent, "id");
-            if (cJSON_IsNumber(json_id)) {
-                //Get ID
-                int agent_id = json_id->valueint;
-
-                //Get labels if any
-                cJSON* json_labels = wdb_global_get_agent_labels(wdb, agent_id);
-                if (json_labels) {
-                    if (json_labels->child) {
-                        cJSON_AddItemToObject(json_agent, "labels", json_labels);
-                    }
-                    else {
-                        cJSON_Delete(json_labels);
+    int initial_agent_id = *last_agent_id;
+
+    for (size_t i = 0; i < sizeof(stmts)/sizeof(*stmts); ++i) {
+        int stmt_id = stmts[i];
+
+        *last_agent_id = initial_agent_id;
+        status = WDBC_UNKNOWN;
+
+        while (status == WDBC_UNKNOWN) {
+            //Prepare SQL query
+            if (wdb_stmt_cache(wdb, stmt_id) < 0) {
+                mdebug1("Cannot cache statement");
+                snprintf(*output, WDB_MAX_RESPONSE_SIZE, "%s", "Cannot cache statement");
+                status = WDBC_ERROR;
+                break;
+            }
+            agent_stmt = wdb->stmt[stmt_id];
+            if (sqlite3_bind_int(agent_stmt, 1, *last_agent_id) != SQLITE_OK) {
+                merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                snprintf(*output, WDB_MAX_RESPONSE_SIZE, "%s", "Cannot bind sql statement");
+                status = WDBC_ERROR;
+                break;
+            }
+
+            //Get agent info
+            cJSON* sql_agents_response = wdb_exec_stmt(agent_stmt);
+            if (sql_agents_response && sql_agents_response->child) {
+                cJSON* json_agent = sql_agents_response->child;
+                cJSON* json_id = cJSON_GetObjectItem(json_agent, "id");
+                if (cJSON_IsNumber(json_id)) {
+                    //Get ID
+                    int agent_id = json_id->valueint;
+
+                    if (stmt_id == WDB_STMT_GLOBAL_SYNC_REQ_FULL_GET) {
+                        //Get labels if any
+                        cJSON* json_labels = wdb_global_get_agent_labels(wdb, agent_id);
+                        if (json_labels) {
+                            if (json_labels->child) {
+                                cJSON_AddItemToObject(json_agent, "labels", json_labels);
```

Comment:
In case cJSON_AddItemToObject fails you could control the removal of the json.
```suggestion
                                if (!cJSON_AddItemToObject(json_agent, "labels", json_labels)) {
                                merror("Failed to add labels to JSON object for agent %d", agent_id);
                                cJSON_Delete(json_labels);
```

Reply:
This function returns void, I'm not sure if there is a way to validate this case. However, I didn't change this code so it is the same as always.